rewrite-core/src/main/java/org/openrewrite/text/DockerImageReference.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.text;
+
+import lombok.Value;
+import org.openrewrite.Cursor;
+import org.openrewrite.SourceFile;
+import org.openrewrite.trait.Reference;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+@Value
+public class DockerImageReference implements Reference {
+    Cursor cursor;
+    String value;
+
+    @Override
+    public Kind getKind() {
+        return Kind.IMAGE;
+    }
+
+    public static class Provider implements Reference.Provider {
+        private static final Pattern FROM = Pattern.compile("FROM\\s+([\\w:.-]*)", Pattern.CASE_INSENSITIVE);
```

Comment:
AFAIK the directives are case sensitive and I added some more allowed characters (possibly more exist).

```suggestion
        private static final Pattern FROM = Pattern.compile("\\bFROM\\s+([\\w/:.-]+)");
```

I also wonder what about the `--platform` option and other places images can occur as in the `COPY` directive's `--from` option.

Reply:
> AFAIK the directives are case sensitive

No, there are not 😉. See also: https://docs.docker.com/reference/build-checks/from-as-casing/.

> I also wonder what about the --platform option and other places images can occur as in the COPY directive's --from option.

I will think if I can implement this as well. Or at least, I will give it a try! I was not even aware you could load images that way 😇.

Reply:
Updated the code to support `--platform` and `--from` options as well. In the end I ditched the regex, doing it manually by going through the entire file gave me more flexibility 💪🏻.